[19.09] Polish YAML based job configuration for 19.09 release.

jmchilton opened the pull request:
Translate much more of the XML advanced configuration docs to job_conf.sample_advanced.yml and fill in small functionality holes missing from along the way.

See individual commits for details - builds on #8614 #8609.

See https://github.com/galaxyproject/galaxy/pull/7854 for the original PR adding YAML based job configuration to 19.09.

jmchilton committed:
Translate embedded Pulsar job docs to advanced YAML sample.

jmchilton committed:
Translate Kubernetes job docs to advanced YAML sample.

jmchilton committed:
Translate custom job metrics job docs to advanced YAML sample.

Document and test.

jmchilton committed:
Cleanup docker config options in XML.

jmchilton committed:
Translate Docker job docs to advanced YAML sample.

jmchilton committed:
Translate Signularity job docs to advanced YAML sample.

jmchilton committed:
Remove reference to wiki documentation I can't actually find.

jmchilton committed:
Translate Condor job docs to advanced YAML sample.

jmchilton committed:
Cleanup resubmit description in XML for recent changes.

jmchilton committed:
Translate resubmit and container runner job docs to advanced YAML sample.

bgruening commented on test/unit/jobs/job_conf.sample_advanced.yml:
```diff
@@ -838,6 +837,102 @@ execution:
+      native_specification: '--mem-per-cpu=512'
+      resubmit:
+      - condition: memory_limit_reached
+        environment: bigmem
```
environment `bigmem` is missing, which might be confusing for users

jmchilton replied:
I did this both in the example XML and the example YAML. Thanks for the catch!

jmchilton committed:
Add bigmem to sample job_confs for correctness.

As suggested here - https://github.com/galaxyproject/galaxy/pull/8616#pullrequestreview-287255028

jmchilton committed:
Fix job metrics for recent plugin_config API changes.
